### ISSUE TITLE
error when using "model merger" 

### ISSUE THREAD

**Issue by coconiox2** [colmap/colmap#1054]
error when using model merger
----------------
I used colmap-GUI to do a sparse reconstruction on two data sets with intersecting images.But I had a problem with the model_merger process.

 ./colmap model_merger --input_path1 /home/hadoop/data/1_space --input_path2 /home/hadoop/data/2_space --output_path /home/hadoop/data/fuse

Reconstruction 1
Images: 40
Points: 45907

Reconstruction 2
Images: 40
Points: 52919

Merging reconstructions
F1113 22:32:54.185003 28747 reconstruction.cc:633] Check failed: Image(image_id).Name() == reconstruction.Image(image_id).Name() (DJI_0240.jpg vs. DJI_0228.jpg)
*** Check failure stack trace: ***
    @     0x7fa1e2b1f5cd  google::LogMessage::Fail()
    @     0x7fa1e2b21433  google::LogMessage::SendToLog()
    @     0x7fa1e2b1f15b  google::LogMessage::Flush()
    @     0x7fa1e2b21e1e  google::LogMessageFatal::~LogMessageFatal()
    @           0x683f7e  colmap::Reconstruction::FindCommonRegImageIds()
    @           0x69f8fe  colmap::ComputeAlignmentBetweenReconstructions()
    @           0x68ea3d  colmap::Reconstruction::Merge()
    @           0x5d309f  RunModelMerger()
    @           0x5bb2fe  main
    @     0x7fa1df1c2840  __libc_start_main
    @           0x5c51a9  _start
    @              (nil)  (unknown)
已放弃 (核心已转储)

Another question
----------------
I want to run colmap on the cluster.First, I distributed the image to the child node, and then sent the result back to the master node for merging after sparse reconstruction on the child node.What should I do?

**Comment by tsattler** [comment]
I would assume that the model_merger tries to identify images in the two models based on their image ids, i.e., it assumes that images having the same id are identical. It then checks whether the file names are the same. Since this does not seem to be the case, it crashed. I would assume that you are using different databases in the child node.
Potential solutions are:
* Use the same database in each child node, but only use a subset of all images for reconstruction (the mapper has a flag that lets you select a subset of images for reconstruction).
* Adapt the code of the model_merger to identify corresponding images based on their filename and not their image id.

**Comment by coconiox2** [comment]
Thanks a lot. It works.But I have another question:
If my images are unordered, how should I divide the list of images to make sure that there is enough overlap between the different models?I wonder if Colmap has this function.And what if I could use GPS information to get which images are spatially adjacent?

**Comment by tsattler** [comment]
You could have a look at the hierarchical mapper, which does some form of subdivision.

**Comment by AIBluefisher** [comment]
> Thanks a lot. It works.But I have another question:
> If my images are unordered, how should I divide the list of images to make sure that there is enough overlap between the different models?I wonder if Colmap has this function.And what if I could use GPS information to get which images are spatially adjacent?

A simple solution is to enlarge the completeness ratio or common images number between clusters. However, it's no guarantee
that these common images could registered successfully in local reconstruction. Instead, I used common 3D landmarks to merge different clusters together. You can also found the code in my repo: [DAGSfM](https://github.com/AIBluefisher/DAGSfM/blob/dev/src/controllers/sfm_aligner.cpp)

Besides, according to my experiment, using GPS for image clustering is not a good idea sometimes. The number of inlier matches is the relatively reasonable weight for graph cut.

**Comment by jeojeda** [comment]
Hello. Is it possible to merge models using the COLMAP.bat portable version?
I can't find the option.

**Comment by tsattler** [comment]
Yes, if you call COLMAP.bat from the power shell with the `model_merger` option.
